flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/service/AbstractFlinkServiceTest.java

```diff
@@ -673,9 +693,11 @@ public void nativeSavepointFormatTest() throws Exception {
                         .set(OPERATOR_SAVEPOINT_FORMAT_TYPE, SavepointFormatType.NATIVE),
                 false);
         assertTrue(stopWithSavepointFuture.isDone());
-        assertEquals(jobID, stopWithSavepointFuture.get().f0);
-        assertEquals(SavepointFormatType.NATIVE, stopWithSavepointFuture.get().f1);
-        assertEquals(savepointPath, stopWithSavepointFuture.get().f2);
+        if (!failAfterSavepointCompletes) {
+            assertEquals(jobID, stopWithSavepointFuture.get().f0);
+            assertEquals(SavepointFormatType.NATIVE, stopWithSavepointFuture.get().f1);
+            assertEquals(savepointPath, stopWithSavepointFuture.get().f2);
+        }
```

Comment:
Am I missing something here, or why are we not validating the savepoint path in both the failure and non failure case?

Reply:
The test asserts on the savepoint future which is an implementation detail which the test really shouldn't assert on. That's testing test assertions but not the production code itself. You are right though, that we shouldn't completely skip this test. I've changed the test to assert on the actual persisted savepoint info.